Author: amesgen <amesgen@amesgen.de>
Date:   Thu Feb 11 17:56:33 2021 +0100

    GHC 9.0 compat w/o tests (#130)
    
    * update to GHC 9.0
    
    * move CI to GH actions

diff --git a/generic-lens.cabal b/generic-lens.cabal
index b88d365..e489ab4 100644
--- a/generic-lens.cabal
+++ b/generic-lens.cabal
@@ -14,7 +14,7 @@ maintainer:           kiss.csongor.kiss@gmail.com
 category:             Generics, Records, Lens
 build-type:           Simple
 cabal-version:        >= 1.10
-Tested-With:          GHC == 8.4.1, GHC == 8.4.4, GHC == 8.6.5, GHC == 8.8.1, GHC == 8.10.1
+Tested-With:          GHC == 8.4.1, GHC == 8.4.4, GHC == 8.6.5, GHC == 8.8.4, GHC == 8.10.3
 
 extra-source-files:   examples/StarWars.hs
                     , examples/Examples.hs
diff --git a/src/Data/Generics/Internal/VL/Iso.hs b/src/Data/Generics/Internal/VL/Iso.hs
index 777d985..eba22f2 100644
--- a/src/Data/Generics/Internal/VL/Iso.hs
+++ b/src/Data/Generics/Internal/VL/Iso.hs
@@ -55,7 +55,7 @@ fromIso l = withIso l $ \ sa bt -> iso bt sa
 {-# inline fromIso #-}
 
 iso2isovl :: P.Iso s t a b -> Iso s t a b
-iso2isovl _iso = P.withIso _iso iso
+iso2isovl _iso = P.withIso _iso $ \ sa bt -> iso sa bt
 {-# INLINE iso2isovl #-}
 
 -- | Extract the two functions, one from @s -> a@ and
diff --git a/src/Data/Generics/Internal/VL/Prism.hs b/src/Data/Generics/Internal/VL/Prism.hs
index 6a4a252..d6cf53c 100644
--- a/src/Data/Generics/Internal/VL/Prism.hs
+++ b/src/Data/Generics/Internal/VL/Prism.hs
@@ -50,7 +50,7 @@ prism bt seta eta = P.dimap (\x -> P.left' pure (seta x)) (either id (\x -> fmap
 {-# INLINE prism #-}
 
 prism2prismvl :: P.APrism i s t a b -> Prism s t a b
-prism2prismvl  _prism = P.withPrism _prism prism
+prism2prismvl  _prism = P.withPrism _prism $ \ bt sta -> prism bt sta
 {-# INLINE prism2prismvl #-}
 
 --------------------------------------------------------------------------------
